src/config_manager.cc

```diff
@@ -106,30 +107,18 @@ void ConfigManager::init()
     options = Ref<Array<ConfigOption> >(new Array<ConfigOption>(CFG_MAX));
 
     String home = userhome + DIR_SEPARATOR + config_dir;
-    bool home_ok = true;
 
     if (filename == nullptr) {
         // No config file path provided, so lets find one.
-        if (home_ok && (!check_path(userhome + DIR_SEPARATOR + config_dir + DIR_SEPARATOR + DEFAULT_CONFIG_NAME))) {
-            home_ok = false;
-        } else {
+        if (check_path(home + DIR_SEPARATOR + DEFAULT_CONFIG_NAME)) {
```

Comment:
All we do now is:

1. Check if the default path is OK
2. Load default path OR throw exception